test_automap.py

```diff
@@ -78,3 +78,11 @@ def test_issue_3(keys: Keys) -> None:
     a |= (key,)
     with pytest.raises(ValueError):
         a |= (key,)
+
+
+def test_non_unique_exception():
+    with pytest.raises(ValueError):
+        _ = automap.AutoMap(("a", "b", "a"))
+
+    with pytest.raises(automap.NonUniqueError):
+        _ = automap.AutoMap(("a", "b", "a"))
```

Comment:
I'd like to hypothesize this, if possible!

We can follow the pattern in the existing tests. Something like this:

```suggestion
@hypothesis.given(keys=hypothesis.infer)
def test_non_unique_exception(keys: Keys) -> None:
    hypothesis.assume(keys)
    duplicate = next(iter(keys))
    with pytest.raises(ValueError):
        automap.AutoMap([*keys, duplicate])

    with pytest.raises(automap.NonUniqueError):
        automap.AutoMap([*keys, duplicate])
```

Reply:
Agreed, this is much better.